pandas/tests/window/moments/test_moments_expanding.py

```diff
@@ -0,0 +1,318 @@
+        self, func, static_comp, has_min_periods=True, preserve_nan=True
+    ):
+
+        series_result = func(self.series)
+        assert isinstance(series_result, Series)
+        frame_result = func(self.frame)
+        assert isinstance(frame_result, DataFrame)
+
+        result = func(self.series)
+        tm.assert_almost_equal(result[10], static_comp(self.series[:11]))
+
+        if preserve_nan:
+            assert result.iloc[self._nan_locs].isna().all()
+
+        ser = Series(randn(50))
+
+        if has_min_periods:
+            result = func(ser, min_periods=30)
+            assert result[:29].isna().all()
+            tm.assert_almost_equal(result.iloc[-1], static_comp(ser[:50]))
+
+            # min_periods is working correctly
+            result = func(ser, min_periods=15)
+            assert isna(result.iloc[13])
+            assert notna(result.iloc[14])
+
+            ser2 = Series(randn(20))
+            result = func(ser2, min_periods=5)
+            assert isna(result[3])
+            assert notna(result[4])
+
+            # min_periods=0
+            result0 = func(ser, min_periods=0)
+            result1 = func(ser, min_periods=1)
+            tm.assert_almost_equal(result0, result1)
+        else:
+            result = func(ser)
+            tm.assert_almost_equal(result.iloc[-1], static_comp(ser[:50]))
+
+    @pytest.mark.parametrize(
+        "f",
+        [
+            lambda x: x.expanding().count(),
+            lambda x: x.expanding(min_periods=5).cov(x, pairwise=False),
+            lambda x: x.expanding(min_periods=5).corr(x, pairwise=False),
+            lambda x: x.expanding(min_periods=5).max(),
+            lambda x: x.expanding(min_periods=5).min(),
+            lambda x: x.expanding(min_periods=5).sum(),
+            lambda x: x.expanding(min_periods=5).mean(),
+            lambda x: x.expanding(min_periods=5).std(),
+            lambda x: x.expanding(min_periods=5).var(),
+            lambda x: x.expanding(min_periods=5).skew(),
+            lambda x: x.expanding(min_periods=5).kurt(),
+            lambda x: x.expanding(min_periods=5).quantile(0.5),
+            lambda x: x.expanding(min_periods=5).median(),
+            lambda x: x.expanding(min_periods=5).apply(sum, raw=False),
+            lambda x: x.expanding(min_periods=5).apply(sum, raw=True),
+        ],
+    )
+    def test_moment_functions_zero_length(self, f):
+        # GH 8056
+        s = Series(dtype=np.float64)
+        s_expected = s
+        df1 = DataFrame()
+        df1_expected = df1
+        df2 = DataFrame(columns=["a"])
+        df2["a"] = df2["a"].astype("float64")
+        df2_expected = df2
+
+        try:
+            s_result = f(s)
+            tm.assert_series_equal(s_result, s_expected)
+
+            df1_result = f(df1)
+            tm.assert_frame_equal(df1_result, df1_expected)
+
+            df2_result = f(df2)
+            tm.assert_frame_equal(df2_result, df2_expected)
+        except (ImportError):
```

Comment:
Instead could you use a `pytest.param(..., mark=pytest.mark.skipif(...)` for the parameter that needs scipy. 

Also, shouldn't all these parameters operate without scipy?

Reply:
yeah, they could operate without scipy. I removed.